app/services/attribute_asserter.rb

```diff
@@ -79,11 +80,19 @@ def add_verified_at(attrs)
   end
 
   def add_aal(attrs)
-    context = authn_request.requested_aal_authn_context
-    context ||= Saml::Idp::Constants::AUTHN_CONTEXT_AAL_TO_CLASSREF[service_provider.aal]
+    requested_context = authn_request.requested_aal_authn_context
+    requested_aal_level = Saml::Idp::Constants::AUTHN_CONTEXT_CLASSREF_TO_AAL[requested_context]
+    aal_level = requested_aal_level || service_provider.aal || ::Idp::Constants::DEFAULT_AAL
```

Comment:
Just to double-check, do we need any special checks here to ensure that, if given, the `requested_all_level` is the highest of the 3 options here? i.e. the use-case that it could be requested as AAL3 for a service provider configured at AAL2.

Reply:
I believe the configured AAL level would be the minimum, so requesting 3 would be ok for an SP configured for 2. In any case, we've talked about removing the SP configuration of AAL level altogether.